### DISCUSSION TITLE
No public key specified when add network using invite

### DISCUSSION THREAD

**Discussion by selankon** [pacien/tincapp#113] [Q&A]
I created an invite for a host from my Tinc `1.1pre18` server, and then manually added an Ip to the invite file. 

On the Android app I added the invite and the network created successfully, but when I check the log, the connection with the server is unsuccessful. 

On the server side the error message is `No public key for android specified!` (where `android` is the android client host name).

Checking on the server side, the `hosts/android` file has only the `Ed25519PublicKey`, instead, on the Android device, there are a `BEGIN RSA PUBLIC KEY` block. 

Is this normal? Is an error implementing the protocol of key exchange? Or is a Tinc bug on the server side? 

Pasting the `RSA` block on the server side, it can't connect neither. This time, showing up the Android log, the error message is `Could not set up a meta connection to server` and `server tries to rollback protocol version 17.0`

Thanks in advance!

**Comment by pacien** [comment]
If I recall correctly, the RSA keypair is only used for the legacy protocol
(tinc 1.0). Tinc 1.1 should be able to connect with only the Ed25519 keys.

The message about protocol rollback seems to indicate that it's trying to use
the legacy protocol instead of the new one.

You can perhaps check that you have the Ed25519 keys on both nodes.
